cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -169,7 +216,15 @@ func main() {
 				if len(unschedulablePodsToHelp) == 0 {
 					glog.V(1).Info("No unschedulable pods")
 				} else {
+					scaleUpStart := time.Now()
+					lastTimestamp.WithLabelValues("scale_up").Set(float64(scaleUpStart.Unix()))
 					scaledUp, err := ScaleUp(unschedulablePodsToHelp, nodes, migConfigs, gceManager, kubeClient, predicateChecker, recorder)
+
+					duration.WithLabelValues("scale_up").Observe(
```

Comment:
Can you opaque these 4 lines into function? The same lines (that different only with the string and start time) you have in 4 different places